tiledb/sm/serialization/consolidation.cc

```diff
@@ -71,34 +72,67 @@ class ConsolidationSerializationDisabledException
 
 #ifdef TILEDB_SERIALIZATION
 
-Status array_consolidation_request_to_capnp(
+void array_consolidation_request_to_capnp(
     const Config& config,
+    const std::vector<std::string>* fragment_uris,
     capnp::ArrayConsolidationRequest::Builder*
         array_consolidation_request_builder) {
+  // Validate input arguments to be sure that what we are serializing make sense
+  auto mode = Consolidator::mode_from_config(config);
+  if (mode != ConsolidationMode::FRAGMENT &&
+      (fragment_uris != nullptr && !fragment_uris->empty())) {
+    throw ConsolidationSerializationException(
+        "[array_consolidation_request_to_capnp] Error serializing "
+        "consolidation request. A non-empty fragment list should only be "
+        "provided for fragment consolidation.");
+  }
+
   auto config_builder = array_consolidation_request_builder->initConfig();
-  RETURN_NOT_OK(config_to_capnp(config, &config_builder));
-  return Status::Ok();
+  throw_if_not_ok(config_to_capnp(config, &config_builder));
+
+  if (fragment_uris != nullptr && !fragment_uris->empty()) {
+    auto fragment_list_builder =
+        array_consolidation_request_builder->initFragments(
+            fragment_uris->size());
+    for (size_t i = 0; i < fragment_uris->size(); i++) {
+      fragment_list_builder.set(i, (*fragment_uris)[i]);
```

Comment:
those need to become relative to cross the wire, otherwise it's a security concern. Use `serialize_array_uri_to_relative` and `deserialize_array_uri_to_absolute` utility functions